File: app/src/main/java/com/antonchaynikov/triptracker/history/HistoryComponent.java
package com.antonchaynikov.triptracker.history;

import com.antonchaynikov.core.viewmodel.CommonViewModelModule;
import com.antonchaynikov.core.viewmodel.CommonViewModelScope;
import com.antonchaynikov.triptracker.injection.AppComponent;

import dagger.Component;

@CommonViewModelScope
@Component(dependencies = AppComponent.class, modules = {HistoryModule.class, CommonViewModelModule.class})
public interface HistoryComponent {
    void inject(HistoryFragment fragment);
}

File: app/src/main/java/com/antonchaynikov/triptracker/HistoryComponent.kt
package com.antonchaynikov.triptracker

import com.antonchaynikov.tripshistory.HistoryFragment
import com.antonchaynikov.tripshistory.HistoryScope
import com.antonchaynikov.tripshistory.TripHistoryModule
import dagger.BindsInstance
import dagger.Component
import dagger.Subcomponent

@HistoryScope
@Subcomponent(modules = [TripHistoryModule::class])
interface HistoryComponent {
    fun inject(historyFragment: HistoryFragment)

    @Subcomponent.Builder
    interface Builder {
        fun build(): HistoryComponent
        @BindsInstance
        fun fragment(fragment: HistoryFragment): Builder
    }
}